File: Python/8py 2/if.py
"""
|--------------------------------------------------------------------------
| The 'if' function
|--------------------------------------------------------------------------
|
"""

def _if(bool, func1, func2):
   return func1() if bool else func2()


def _if(bool, func1, func2):
	return [func2, func1][bool]()


def _if(bool, func1, func2):
	return (bool and func1 or func2)()


def truthy(): 
	print("True")
def falsey(): 
	print("False")
print(_if(True, truthy, falsey))

File: JavaScript/8js 2/if.js
/*
|--------------------------------------------------------------------------
| The 'if' function
|--------------------------------------------------------------------------
|
*/

function _if(bool, func1, func2) {
   return bool ? func1() : func2();
}


function _if(bool, func1, func2) {
   return arguments[~bool + 3]();
}


function _if(bool, func1, func2) {
   return (bool ? func1 : func2)();
}


console.log(
   _if(
      true,
      function () {
         console.log("True");
      },
      function () {
         console.log("false");
      }
   )
);